### PR TITLE
add docs to the envs aspect

### PR DESCRIPTION
**Pull request by giteden**
## Proposed Changes

-
-
-

### CODE REVIEW

File: scopes/envs/envs/envs.docs.md [hunk 1]
```diff
@@ -1,33 +1,797 @@
...
+}
+```
+
+> A single component (with the same version) cannot use more than a single environment.
+
+### Setting multiple environments
+
+A single workspace can use different environments for different sets of components. Setting an environment on a specific group of components is done by selecting the group and applying the environment. This is done using `teambit.workspace/variants`. To learn more about using 'variant' to select components, [see here](/docs/workspace/cascading-rules)
+
+For example, to set the Node and React environments on two sets of components (selected by their directory):
```
Comment (GiladShoham) [inline scopes/envs/envs/envs.docs.md:126]: you should talk here about setting the `eambit.envs/envs`
consider the following ws.jsonc
```json
{
  "teambit.workspace/variants": {
    "*": {
      "teambit.react/react": {}
    },
    "components/utils": {
      "teambit.harmony/node": {}
    }
  }
}
```
With this config (which will be quite popular) every component under utils will get both node and react as aspects.
There is no guarantee in that case that the env will be used for utils is indeed node.
to solve it, you should add the `teambit.envs/envs` like this:
```json
{
  "teambit.workspace/variants": {
    "*": {
      "teambit.react/react": {}
    },
    "components/utils": {
      "teambit.harmony/node": {},
      "teambit.envs/envs": {
         "env": "teambit.harmony/node"
       }
    }
  }
}
```

File: scopes/envs/envs/envs.docs.md [hunk 2]
```diff
@@ -1,33 +1,797 @@
...
+- The **"base" environment** (e.g, `@teambit/react`) is extended and customized using its override methods. Each override method, or "environment transformer", corresponds to a Bit extension component used by the environment (e.g, the TypeScript component). Using an 'environment transformer' will add new configurations to the relevant Bit component and will override any conflicting ones.<br /> The full list of available 'environment transformers' can be seen in the specific environment's documentation (see: React, React Native, Node).
+
+- The **'Environments' component** (`@teambit/envs`) is used to:
+  1. Register the new environment using its [slot](TODO)
+  2. Override a ["service handler"](TODO). This is done to replace a Bit component used by an environment service. For example, to set the "compiler" service handler to use Babel instead of TypeScript (see an example, [here](/docs/environments/build-environment#override-a-service-handler)).
+
+#### Override the config for a Bit component used by the environment
+
+The example below is of a React environment extension. This new environment overrides React's DevServer configuration by setting a new Webpack configuration file.
```
Comment (GiladShoham) [inline scopes/envs/envs/envs.docs.md:251]: consider mention that those APIs are about to be deprecated / replaced soon

File: scopes/envs/envs/envs.docs.md [hunk 3]
```diff
@@ -1,33 +1,797 @@
...
+
+For example:
+```shell
+$ touch path/to/extension/env-extension.aspect.ts
+```
+
+Place the following lines to register your environment as a multiple runtime extension (a.k.a, an Aspect):
+
+```ts
+// env-extension.ts
```
Comment (GiladShoham) [inline scopes/envs/envs/envs.docs.md:686]: env-extension.aspect.ts

File: scopes/envs/envs/envs.docs.md [hunk 4]
```diff
@@ -1,33 +1,797 @@
...
+
+### Registering a runtime extension
+An aspect is a collection of multiple extensions, each extending a specific runtime.
+
+Register each runtime extension to its corresponding runtime, using the `addRuntime` method.
+
+For example:
+
+```typescript
+// react-with-providers.preview.ts
```
Comment (GiladShoham) [inline scopes/envs/envs/envs.docs.md:706]: inconsistent with the example above - 
above you used env-extesnion.ts in the comment and here the react-with-providers
make all consistent

File: scopes/envs/envs/envs.docs.md [hunk 5]
```diff
@@ -1,33 +1,797 @@
...
+
```
Comment (GiladShoham) [inline scopes/envs/envs/envs.docs.md:707]: missing importing the `ReactWithProvidersAspect`

File: scopes/envs/envs/envs.docs.md [hunk 6]
```diff
@@ -1,33 +1,797 @@
...
+    const newReactEnv = react.compose([
+      react.overrideTsConfig(tsconfig)
+    ]);
+
+    envs.registerEnv(newReactEnv);
+
+    return new ReactExtensionMain(react, envs);
+
 }
+  ReactExtensionMain.addRuntime(ReactExtensionMain);
```
Comment (GiladShoham) [inline scopes/envs/envs/envs.docs.md:760]: should be `ReactWithProvidersAspect.addRuntime`

File: scopes/envs/envs/envs.docs.md [hunk 7]
```diff
@@ -1,33 +1,797 @@
...
+Node files that run in a node runtime environments and outputs to the terminal.
+
+**Example:**  
+The React environment TypeScript compiler will be extended in the main runtime.
+
+```typescript
+// react-extension.main.ts
+
+const tsconfig = require('./typescript/tsconfig.json');
+
```
Comment (GiladShoham) [inline scopes/envs/envs/envs.docs.md:741]: missing importing `ReactWithProvidersAspect`

**Review by GiladShoham — CHANGES_REQUESTED** [review]

File: scopes/envs/envs/envs.docs.md (hunk 7 above)
Comment (giteden) [reply scopes/envs/envs/envs.docs.md:741]: Done.

File: scopes/envs/envs/envs.docs.md (hunk 6 above)
Comment (giteden) [reply scopes/envs/envs/envs.docs.md:760]: Done.

File: scopes/envs/envs/envs.docs.md (hunk 5 above)
Comment (giteden) [reply scopes/envs/envs/envs.docs.md:707]: Done.

File: scopes/envs/envs/envs.docs.md (hunk 4 above)
Comment (giteden) [reply scopes/envs/envs/envs.docs.md:706]: I aimed for two independent examples but I can see how that can be confusing. (Done.)

File: scopes/envs/envs/envs.docs.md (hunk 3 above)
Comment (giteden) [reply scopes/envs/envs/envs.docs.md:686]: Done.

File: scopes/envs/envs/envs.docs.md (hunk 2 above)
Comment (giteden) [reply scopes/envs/envs/envs.docs.md:251]: Done.

File: scopes/envs/envs/envs.docs.md (hunk 1 above)
Comment (giteden) [reply scopes/envs/envs/envs.docs.md:126]: Done (see the warning in the first WS config section + the new 'troubleshooting' section)

**Pull request merged**